pkg/csv/reader.go

```diff
@@ -45,16 +45,48 @@ func (r *CSVReader) ReadLine() (base.Data, error) {
 		}
 	}
 
+	lineBytesNum := countLineBytes(line)
 	if *r.CSVConfig.WithLabel {
 		switch line[0] {
 		case "+":
-			return base.InsertData(line[1:]), nil
+			return base.InsertData(line[1:], lineBytesNum), nil
 		case "-":
-			return base.DeleteData(line[1:]), nil
+			return base.DeleteData(line[1:], lineBytesNum), nil
 		default:
 			return base.Data{}, fmt.Errorf("Invalid label: %s", line[0])
 		}
 	} else {
-		return base.InsertData(line), nil
+		return base.InsertData(line, lineBytesNum), nil
 	}
 }
+
+func countLineBytes(record []string) int {
+	// comma and '\n'
+	charNum := len(record)
+	bytesNum := charNum
+	for _, s := range record {
+		bytesNum += len(s)
+	}
+	return bytesNum
+}
+
+func CountFileBytes(path string, withHeader bool) (int64, error) {
+	file, err := os.Open(path)
```

Comment:
The statistical methods of the two functions are different.

- `countLineBytes` is the sum of bytes of the csv field.
- `CountFileBytes` is to get the file size and then subtract the header bytes.

Reply:
```golang


var _ io.Reader = (*recordReader)(nil)

type recordReader struct {
	io.Reader
	lastReadBytes int
}

func (r *recordReader) Read(p []byte) (n int, err error) {
	n, err = r.Reader.Read(p)
	r.lastReadBytes = n
	return
}
func (r *CSVReader) InitReader(file *os.File) {
	r.rr = &recordReader{Reader: bufio.NewReader(file)}
	r.reader = csv.NewReader(r.rr)
	...
}

func (r *CSVReader) ReadLine() (base.Data, error) {
    line, err := r.reader.Read()
    // r.rr.lastReadBytes // get the last read bytes 
    ...
}
```